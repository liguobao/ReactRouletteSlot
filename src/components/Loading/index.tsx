/*
 * @Author: wzi
 * @Date: 2018-03-02 10:04:09
 * @Last Modified by: wzi
 * @Last Modified time: 2018-04-25 15:56:27
 */

import ErrorBoundary from '@common/decorator/errorBoundaryDecorator';
import React from 'react';
import styled from '@common/helper/styled-component';
import ReactLoading from 'react-loading';
const width = 80;
const height = 40;
const Wrapper = styled.div`
    position: fixed;
    top: calc(50% - ${height / 2}px);
    left: calc(50% - ${width / 2}px);
    width: ${width}px;
    height: ${height}px;
`;
type LoadProps = {};
type LoadState = {};
@ErrorBoundary('Loading 加载失败')
export default class Load extends React.PureComponent<LoadProps, LoadState> {
    render() {
        return (
            <Wrapper>
                <ReactLoading
                    type="bars"
                    height={height}
                    width={width}
                    color="#ed6c44"
                />
            </Wrapper>
        );
    }
}
