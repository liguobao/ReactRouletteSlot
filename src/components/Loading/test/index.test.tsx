/*
 * @Author: wzi
 * @Date: 2018-01-31 16:40:02
 * @Last Modified by: wzi
 * @Last Modified time: 2018-04-27 15:32:26
 */
import React from 'react';
import Loading from '@components/Loading';
import { mount } from 'enzyme';
import 'jest-styled-components';
import toJson from 'enzyme-to-json';
describe('Loading', () => {
    let _wrapper;
    beforeEach(() => {
        _wrapper = mount(<Loading />);
    });
    test('整体', () => {
        expect(toJson(_wrapper)).toMatchSnapshot();
    });
});
