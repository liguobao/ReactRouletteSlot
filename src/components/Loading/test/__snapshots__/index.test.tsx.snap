// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading 整体 1`] = `
.c0 {
  position: fixed;
  top: calc(50% - 20px);
  left: calc(50% - 40px);
  width: 80px;
  height: 40px;
}

<Load>
  <styled.div>
    <div
      className="c0"
    >
      <Loading
        color="#ed6c44"
        delay={0}
        height={40}
        type="bars"
        width={80}
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 32 32\\">
  <path transform=\\"translate(2)\\" d=\\"M0 12 V20 H4 V12z\\"> 
    <animate attributeName=\\"d\\" values=\\"M0 12 V20 H4 V12z; M0 4 V28 H4 V4z; M0 12 V20 H4 V12z; M0 12 V20 H4 V12z\\" dur=\\"1.2s\\" repeatCount=\\"indefinite\\" begin=\\"0\\" keytimes=\\"0;.2;.5;1\\" keySplines=\\"0.2 0.2 0.4 0.8;0.2 0.6 0.4 0.8;0.2 0.8 0.4 0.8\\" calcMode=\\"spline\\"  />
  </path>
  <path transform=\\"translate(8)\\" d=\\"M0 12 V20 H4 V12z\\">
    <animate attributeName=\\"d\\" values=\\"M0 12 V20 H4 V12z; M0 4 V28 H4 V4z; M0 12 V20 H4 V12z; M0 12 V20 H4 V12z\\" dur=\\"1.2s\\" repeatCount=\\"indefinite\\" begin=\\"0.2\\" keytimes=\\"0;.2;.5;1\\" keySplines=\\"0.2 0.2 0.4 0.8;0.2 0.6 0.4 0.8;0.2 0.8 0.4 0.8\\" calcMode=\\"spline\\"  />
  </path>
  <path transform=\\"translate(14)\\" d=\\"M0 12 V20 H4 V12z\\">
    <animate attributeName=\\"d\\" values=\\"M0 12 V20 H4 V12z; M0 4 V28 H4 V4z; M0 12 V20 H4 V12z; M0 12 V20 H4 V12z\\" dur=\\"1.2s\\" repeatCount=\\"indefinite\\" begin=\\"0.4\\" keytimes=\\"0;.2;.5;1\\" keySplines=\\"0.2 0.2 0.4 0.8;0.2 0.6 0.4 0.8;0.2 0.8 0.4 0.8\\" calcMode=\\"spline\\" />
  </path>
  <path transform=\\"translate(20)\\" d=\\"M0 12 V20 H4 V12z\\">
    <animate attributeName=\\"d\\" values=\\"M0 12 V20 H4 V12z; M0 4 V28 H4 V4z; M0 12 V20 H4 V12z; M0 12 V20 H4 V12z\\" dur=\\"1.2s\\" repeatCount=\\"indefinite\\" begin=\\"0.6\\" keytimes=\\"0;.2;.5;1\\" keySplines=\\"0.2 0.2 0.4 0.8;0.2 0.6 0.4 0.8;0.2 0.8 0.4 0.8\\" calcMode=\\"spline\\" />
  </path>
  <path transform=\\"translate(26)\\" d=\\"M0 12 V20 H4 V12z\\">
    <animate attributeName=\\"d\\" values=\\"M0 12 V20 H4 V12z; M0 4 V28 H4 V4z; M0 12 V20 H4 V12z; M0 12 V20 H4 V12z\\" dur=\\"1.2s\\" repeatCount=\\"indefinite\\" begin=\\"0.8\\" keytimes=\\"0;.2;.5;1\\" keySplines=\\"0.2 0.2 0.4 0.8;0.2 0.6 0.4 0.8;0.2 0.8 0.4 0.8\\" calcMode=\\"spline\\" />
  </path>
</svg>
",
            }
          }
          style={
            Object {
              "fill": "#ed6c44",
              "height": 40,
              "width": 80,
            }
          }
        />
      </Loading>
    </div>
  </styled.div>
</Load>
`;
